thewalrus/libwalrus.pyx

```diff
@@ -468,5 +573,185 @@ def renorm_hermite_multidimensional_real(double [:, :] R, double [:] y, int cuto
 
     for i in range(n):
         y_mat.push_back(y[i])
+    length = cutoff**n
+    cdef double *array = renorm_hermite_multidimensional_cpp(R_mat, y_mat, cutoff)
+
+    return double_pointer_to_array(array, length)
+
+
+def interferometer(double complex[:, :] R, int cutoff):
+    r"""Returns the matrix elements of an interferometer by using a custom version of
+    the renormalized Hermite polynomials.
+
+    Args:
+        R (array[complex128]): square matrix parametrizing the interferometer
+        cutoff (int): maximum size of the subindices in the tensor
+
+    Returns:
+        array[complex128]: the matrix elements of the interferometer
+    """
+    cdef int i, j, n = R.shape[0]
+    cdef vector[double complex] R_mat
+
+    for i in range(n):
+        for j in range(n):
+            R_mat.push_back(R[i, j])
+
+    length = cutoff**n
+    cdef double complex *array = interferometer_cpp(R_mat, cutoff)

```

Comment:
Oh, I guess this is why it has the _cpp, to distinguish it from this function here?